tests/Meziantou.Analyzer.Test/Rules/UseShellExecuteAnalyzerTests.cs

```diff
@@ -0,0 +1,141 @@
+                                      }
+                                  }
+                                  """;
+        await CreateProjectBuilder()
+            .WithSourceCode(SourceCode)
+            .ShouldReportDiagnosticWithMessage("UseShellExecute must be explicitly set when initializing a ProcessStartInfo")
+            .ValidateAsync();
+    }
+
+    [Fact]
+    public async Task Process_start_should_report_false_positives()
+    {
+        const string SourceCode = """
+                                  using System.Diagnostics;
+
+                                  class TypeName
+                                  {
+                                      public void Test()
+                                      {
+                                          var processStartInfo = [||]new ProcessStartInfo();
+                                          processStartInfo.UseShellExecute = false;
+                                          Process.Start(processStartInfo);
+                                      }
+                                  }
+                                  """;
+        await CreateProjectBuilder()
+            .WithSourceCode(SourceCode)
+            .ShouldReportDiagnosticWithMessage("UseShellExecute must be explicitly set when initializing a ProcessStartInfo")
+            .ValidateAsync();
+    }
+
+    [Fact]
+    public async Task Process_start_should_report_when_use_shell_execute_is_not_set_2()
+    {
+        const string SourceCode = """
+                                  using System.Diagnostics;
+
+                                  class TypeName
+                                  {
+                                      public void Test()
+                                      {
+                                          var processStartInfo = [||]new ProcessStartInfo()
+                                          {
+                                              FileName = "notepad",
+                                          };
+                                          Process.Start(processStartInfo);
+                                      }
+                                  }
+                                  """;
+        await CreateProjectBuilder()
+            .WithSourceCode(SourceCode)
+            .ShouldReportDiagnosticWithMessage("UseShellExecute must be explicitly set when initializing a ProcessStartInfo")
+            .ValidateAsync();
+    }
+
+
+    [Fact]
+    public async Task Process_start_should_report_when_using_overload_with_no_process_start_info()
+    {
+        const string SourceCode = """
+                                  using System.Diagnostics;
+
+                                  class TypeName
+                                  {
+                                      public void Test()
+                                      {
+                                          [||]Process.Start("notepad");
```

Comment:
I'm not sure to understand why it should be reported. `UseShellExecute` is only useful when you want to redirect input/output. This method doesn't redirect output, so why do you need to report it?

Reply:
Because it has a different default value between .NET Framework an .NET core.

Reply:
(common issue when you migrate an app from .NET Framework to modern .NET)